File: src/main/kotlin/ch/barrierelos/backend/model/scanner/AnalysisJob.kt
package ch.barrierelos.backend.model.scanner

import kotlinx.datetime.Clock
import kotlinx.datetime.Instant
import kotlinx.serialization.Serializable

@Serializable
public data class AnalysisJob(
  public val id: Long = 0,
  public val modelVersion: String,
  public val jobTimestamp: Instant = Clock.System.now(),
  public val locale: String,
  public val websiteBaseUrl: String,
  public val webpagePaths: MutableSet<String>,
  public var modified: Long = System.currentTimeMillis(),
  public var created: Long = System.currentTimeMillis(),
)

File: src/__mocks__/analysisJob.ts
import {AnalysisJob} from "../model.js";

function getEmptyAnalysisJob(): AnalysisJob {
    return {
        jobId: 0,
        jobTimestamp: "",
        modelVersion: "",
        locale: "en",
        webpagePaths: [],
        websiteBaseUrl: ""
    }
}

export {getEmptyAnalysisJob}
